src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleThreadContext.java

```diff
@@ -260,6 +317,43 @@ public InterimModule buildModule(@Nullable Registry registry) throws EvalExcepti
       }
       extensionUsages.add(extensionUsageBuilder.buildUsage());
     }
+    // Check for chains in repo overrides: If A is overridden with B, and B is overridden with C,
```

Comment:
I think this could probably be more simply formulated as "a repo cannot both be overriding and overridden". Makes the error reporting a bit cleaner as well.

Reply:
I updated the comment, but I'm not sure how to update the error reporting. I would still like to suggest shortening A -> B -> C to just A -> C and wouldn't know how to do that with the new shorter error message.

Reply:
IMO we don't need to suggest the shortening, but I don't feel strongly about this. So up to you what to do here.